### DISCUSSION TITLE
Unable to locate 'Avalonia.Platform.IWindowingPlatform

### DISCUSSION THREAD

**Discussion by FranklinWu98** [AvaloniaUI/Avalonia#16359] [Q&A] [answered]
![Screenshot 2024-07-17 232343](https://github.com/user-attachments/assets/e491dbe7-ba61-43e3-8a2c-0624c5c0cf9e)
I am trying to embed Avalonia in WPF using a HwndHost for second development of Grasshopper. I am encounter the following problem when i tried to add Avalonia into a WPF window that invoked form another program(Grasshopper). The Avalonia works well with wpf when is invoke independtly, not from the Grasshopper program. Does anyone know how to fix it?

Unable to locate 'Avalonia.Platform.IWindowingPlatform

**Comment by maxkatz6** [comment, marked as answer]
Avalonia needs to be initialized before accessing any of its APIs.
Something like this at the very least:
https://github.com/AvaloniaUI/Avalonia/blob/master/samples/interop/WindowsInteropTest/Program.cs#L17-L20

**Reply by FranklinWu98** [reply]
Thank you!
